packages/sdk/client-services/src/packlets/spaces/data-space-manager.test.ts

```diff
@@ -168,5 +168,50 @@ describe('DataSpaceManager', () => {
         500,
       );
     });
+
+    test('activate opens a lazily loaded space', async () => {
+      const builder = new TestBuilder();
+
+      const peer = builder.createPeer();
+      await peer.createIdentity();
+      await openAndClose(peer.echoHost, peer.dataSpaceManager);
+
+      await peer.dataSpaceManager.createSpace();
+      await reloadDataSpaces(peer);
+
+      const space = getFirstSpace(peer);
+      expect(space.state).to.equal(SpaceState.CLOSED);
+      await space.activate();
+      await asyncTimeout(
+        space.stateUpdate.waitForCondition(() => space.state === SpaceState.READY),
+        500,
+      );
+    });
+
+    test('deactivate lazily loaded space ', async () => {
+      const builder = new TestBuilder();
+
+      const peer = builder.createPeer();
+      await peer.createIdentity();
+      await openAndClose(peer.echoHost, peer.dataSpaceManager);
+
+      await peer.dataSpaceManager.createSpace();
+      await reloadDataSpaces(peer);
+
+      await getFirstSpace(peer).deactivate();
+
+      await reloadDataSpaces(peer);
+
+      expect(getFirstSpace(peer).state).to.eq(SpaceState.INACTIVE);
+    });
   });
+
+  const reloadDataSpaces = async (peer: TestPeer) => {
+    await peer.dataSpaceManager.close();
+    await peer.dataSpaceManager.open();
+  };
+
+  const getFirstSpace = (peer: TestPeer) => {
+    return [...peer.dataSpaceManager.spaces.values()][0];
```

Comment:
just as a code consistency thing:

```suggestion
    return Array.from(peer.dataSpaceManager.spaces.values())[0];
```